### PR TITLE
Use an internal ingress by default when the cluster is marked internal

### PR DESCRIPTION
**Pull request by ironcladlou**
When the cluster is internal according to the install config's publishing field,
render an internal default ingresscontroller so that ingress remains consistent
with the intent for the entire cluster to be "internal".

This commit is part of the "User-defined Default Ingress Controller" enhancement
proposal[1] and implements the final user-facing component of the feature.

[1] https://github.com/openshift/enhancements/blob/master/enhancements/user-defined-default-ingress-controller.md

### CODE REVIEW

File: pkg/asset/manifests/ingress.go [hunk 1]
```diff
@@ -37,38 +41,81 @@ func (*Ingress) Dependencies() []asset.Asset {
...
+			},
+			Spec: operatorv1.IngressControllerSpec{
+				EndpointPublishingStrategy: &operatorv1.EndpointPublishingStrategy{
+					Type: operatorv1.PrivateStrategyType,
+				},
+			},
+		}
+		return yaml.Marshal(obj)
+	default:
+		return []byte{}, nil
```
Comment (abhinavdahiya) [inline pkg/asset/manifests/ingress.go]: should we return `nil, nil`, why return a byte slice of `len 0`

Comment (ironcladlou) [reply pkg/asset/manifests/ingress.go]: Fixed

**Comment by ironcladlou** [comment]
@abhinavdahiya 

I managed to test this on AWS using `upi/aws/cloudformation/01_vpc.yaml` to set up the VPC/subnets and then specifying `publish: Internal` in the install config. I ran the installer build from a bastion host within the VPC and came out the other side with a functional cluster with ingress using a load balancer with `scope: Internal`. Everything seems to be fine.

Hope that's a sufficient test. I can provide more details if necessary!

**Comment by ironcladlou** [comment]
All feedback addressed and verified with a manual test, PTAL.

**Comment by ironcladlou** [comment]
Rebased.

**Comment by abhinavdahiya** [comment]
> @abhinavdahiya
> 
> I managed to test this on AWS using `upi/aws/cloudformation/01_vpc.yaml` to set up the VPC/subnets and then specifying `publish: Internal` in the install config. I ran the installer build from a bastion host within the VPC and came out the other side with a functional cluster with ingress using a load balancer with `scope: Internal`. Everything seems to be fine.
> 
> Hope that's a sufficient test. I can provide more details if necessary!

/lgtm

**Pull request merged**